scripts/build_rust_artifacts.py

```diff
@@ -7,53 +7,88 @@
-    subprocess.check_call(command)
+    print(f"Running: {command}")
+    exit_code = subprocess.check_call(command, shell=True, text=True)
+    assert exit_code == 0
 
 
-def main():
-    parser = argparse.ArgumentParser(description="Builds rust artifacts for CI")
-    parser.add_argument("out_dir", help="Output directory for artifacts")
-    parser.add_argument(
-        "cargo_profile", help="Cargo profile to use for compiling the crates"
-    )
-    parser.add_argument("crates", nargs="+", help="List of crate names to be processed")
+def run_with_stdout(command):
+    assert isinstance(command, str)
+    print(f"Running: {command}")
+    cmd_output = subprocess.check_output(command, shell=True, text=True)
+    return cmd_output
 
-    args = parser.parse_args()
 
-    targets = ["aarch64", "x86_64"]
+def find_cargo_deb_crates():
+    jq_query = (
+        ".workspace_members[] as $wm "
+        "| .packages[ ] "
+        "| .id as $id "
+        "| select( $wm | contains($id)) "
+        '| select( .metadata | has("deb")) '
+        "| .name"
+    )
+    command = f"cargo metadata --format-version=1 | jq '{jq_query}'"
+    cmd_output = subprocess.check_output(command, shell=True, text=True)
+    crates = [c.strip('"') for c in cmd_output.strip().split("\n")]
+    return crates
 
-    targets_option = " ".join([f"--target {t}-unknown-linux-gnu" for t in targets])
-    print(f"TARGETS={targets_option}")
 
-    cmd(
+def build_all_crates(*, cargo_profile, targets):
+    targets_option = " ".join([f"--target {t}-unknown-linux-gnu" for t in targets])
+    run(
         f"cargo zigbuild --all "
-        f"--profile {args.cargo_profile} "
+        f"--profile {cargo_profile} "
         f"{targets_option} "
         f"--no-default-features"
     )
 
-    for b in args.crates:
-        os.makedirs(os.path.join(args.out_dir, b), exist_ok=True)
-        print(f"Creating .deb package for {b}:")
+
+def run_cargo_deb(*, out_dir, cargo_profile, targets, crates):
+    for c in crates:
+        os.makedirs(os.path.join(out_dir, c), exist_ok=True)
+        print(f"Creating .deb package for {c}:")
         for t in targets:
-            cmd(
+            run(
                 f"cargo deb --no-build --no-strip "
-                f"--profile {args.cargo_profile} "
-                f"-p {b} "
+                f"--profile {cargo_profile} "
+                f"-p {c} "
                 f"--target {t}-unknown-linux-gnu "
-                f"-o {args.out_dir}/{b}/{b}_{t}.deb"
+                f"-o {out_dir}/{c}/{c}_{t}.deb"
             )
-            cmd(
+            run(
                 f"cp -L "
-                f"target/{t}-unknown-linux-gnu/{args.cargo_profile}/{b} "
-                f"{args.out_dir}/{b}/{b}_{t}"
+                f"target/{t}-unknown-linux-gnu/{cargo_profile}/{c} "
+                f"{out_dir}/{c}/{c}_{t}"
             )
 
 
+def main():
+    parser = argparse.ArgumentParser(description="Builds rust artifacts for CI")
```

Comment:
very nit for whenever (if) you have time: It would be amazing to have a description that our artifacts are actually debian packages, and also have the description declaring what are the assumptions under which we can build a debian.